src/Interface/Character/Parses.js

```diff
@@ -249,10 +250,12 @@ class Parses extends React.Component {
     if (!realm) {
       console.warn('Realm could not be found: ' + this.props.realm + '. This generally indicates a bug.');
     }
-    const realmSlug = realm ? realm.slug : this.props.realm;
+    this.setState({
```

Comment:
I don't think you should use state for this; the realmSlug/realm is based directly on the value of a property and as such it shouldn't be a part of the state. How about refactoring this method a bit so it's not necessary? For example you could move the code to determine the realm (including the warning) to its own method and then call that here and in the render to get the realm. If you add a second method on top of that to get the slug with the property based fallback, you could also eliminate the code duplication of `realm ? realm.slug : this.props.realm`.

Reply:
This makes sense, I had to leverage off the `componentDidMount` in order to interact with the render so that I didn't have to throw anything into state. Let me know how you feel about an `async` on `componentDidMount`, I imagine it's not the most common thing to do.